### ISSUE TITLE
How to config  multi objective   optimization

### ISSUE THREAD

**Issue by kemites** [facebook/Ax#373]
Hello ,
I'm working on multi objective optimisation  and i m looking  for some  application example code of multi objective optimization  to understand  how can i make my one. I have already understand how to config an single objective  optimization .
thanks you

**Comment by Balandat** [comment]
Hi @kemites, but multi-objective you mean exploring the pareto front rather than adding outcome constraints to a single-objective optimization, right? This functionality is in the works and should go in together with additional documentation and tutorials in the near future (cc @sdaulton , @2timesjay)

**Comment by kemites** [comment]
yes ,
I am  student and for my internship , i am working on multi objective optimization.
actually a want to have one example of  code to use or to adapt it for my problem.
i want some thing like this 
https://ax.dev/docs/api.html

**Comment by lena-kashtelyan** [comment]
HI, @kemites, we unfortunately do not yet have a working example, as this functionality is under construction. @2timesjay should be able to provide one soon, however!

**Comment by Balandat** [comment]
Hi @kemites, we have added functionality for multi-objective optimization, please see the tutorial for details how to set this up: https://ax.dev/tutorials/multiobjective_optimization.html

**Comment by Sanghee-Choi** [comment]
Hello, @Balandat. I'm a data scientist and now trying to use Ax library for a practical problem. Before that, I apply this amazing library to a toy problem. Unfortunately, it's not working. So, I've followed the code that you shared, https://ax.dev/tutorials/multiobjective_optimization.html. But, it doesn't work, either.

**Comment by hanochk** [comment]
Hi as for the example, using get_MOO_EHVI() means it isn't the q version qEHVI which should be the parallel version? how do I get verbose printing of the optimization by get_MOO_EHVI() ? suppose my optimization isn;t based on noisy samples, for having the q version do I have only  get_MOO_NEHVI() which is for noisy examples?

**Comment by Balandat** [comment]
@hanochk, `get_MOO_EHVI` will get the modelbridge using the qEHVI acquisition function (it will work for batch BO with `q>1`). While `get_MOO_NEHVI()` does work better for noisy examples, it also works very well for non-noisy problems and has the additional benefit that it scales much better to larger values of `q` (see our [recent paper](https://papers.nips.cc/paper/2021/hash/11704817e347269b7254e744b5e22dac-Abstract.html) for details).

Basically, the bottom line is that in general you'll want to use qNEHVI rather than qEHVI in pretty much all situations, noisy observations or not.

**Comment by hanochk** [comment]
Hi Max, thanks for your reply. Do you have MO example using many arms for
q>1 for the qNEHVI ? 2 objectives . Thanks

**Comment by hanochk** [comment]
Trying to convert that notebook : /multiobjective_optimization.ipynb using
get_MOO_NEHVI () what has to be done is to increase the batch from 1
as ehvi_model.gen(1)
to many?

**Comment by hanochk** [comment]
Hi Max, the main reason for that notebook running slow is the fact
that:ehvi_model.gen(1) i.e does it work in parallel? . I've replaced the
get_MOO_EHVI with get_MOO_NEHVI but the SOBOL loop for instance runs for 30
min at least. What could be done to understand why it runs slow or how to
add more arms for the parallelism?

**Comment by Balandat** [comment]
So is the issue here that the evaluation of the function with the generated candidate takes a long time (rather than the call to `.gen()`)? Does your implementation parallelize evaluation of multiple candidates easily? 

In order to generate `q` candidates for parallel evaluation on the Ax side, you'll just have to call `.gen(q)` with `q>1` and then evaluate the generated candidates in parallel in your code.

**Comment by hanochk** [comment]
The function evaluation is written in c, doesn't support  parallelism, but
can handle vector of candidates. In that case I can use the gen(n) n
candidates ? It takes 500ms indeed fair amount of time. Thanks
